include/zephyr/drivers/dai.h

```diff
@@ -328,6 +328,8 @@ __subsystem struct dai_driver_api {
 	int (*ts_stop)(const struct device *dev, struct dai_ts_cfg *cfg);
 	int (*ts_get)(const struct device *dev, struct dai_ts_cfg *cfg,
 		      struct dai_ts_data *tsd);
+	int (*dma_control_set)(const struct device *dev, const void *dma_control_params,
+			       size_t size);
```

Comment:
And another: Can this be called in RUNNING state?

Reply:
Now I would like to keep it void. The parameter for this function could be a generic TLV, but I would prefer to do that later during the refactor.

Reply:
I've updated the function documentation to clarify that dai_dma_control_set must only be called when the DAI is in the READY state, as per the DAI driver's operational flow described in the documentation. This ensures DMA control settings are safely applied before streaming begins.

Reply:
@tmleman  If you look at this from generic dai.h point of view, if this is "void*", then this is completely opaque interface, just like the "set_config" now. I'm ok to add such new opaque interface, but we shouldn't call it dma_control if there's nothing about the DMA control in the generic interface. This is really a function to update the config in RUNNING state, and on some Intel DAI drivers, this is used to modify DMA state, but in other DAI implementations, this could be used for something completely different. Makes sense?